File: solutions/0209-minimum-size-subarray-sum/solution.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        m=2147483647
        s=0
        n=len(nums)
        left,right=0,0
        while(right<n):
            s+=nums[right];
            while(s>=target):
                m=min(m,right-left+1)
                s-=nums[left]
                left+=1
            right+=1
        return 0 if m==2147483647 else m

File: solutions/0209-minimum-size-subarray-sum/solution.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        Map<Integer,Integer>map=new HashMap<>();
        int n=nums.length;
        int sum=0,m=Integer.MAX_VALUE;
        for(int left=0,right=0;right<n;right++)
        {
            sum+=nums[right];
            while(sum>=target)
            {
                m=Math.min(m,right-left+1);
                sum-=nums[left++];
            }
        }
        return m==2147483647?0:m;
    }
}
